R/convert_output_types.R

```diff
@@ -55,31 +55,59 @@ convert_output_type <- function(model_out_tbl, terminal_output_type_id) {
 #' @noRd
 #' @importFrom rlang .data
 convert_single_output_type <- function(terminal_output_type, terminal_output_type_id, model_out_tbl) {
-  task_id_cols <- subset_task_id_names(names(model_out_tbl))
+  model_out_cols <- colnames(model_out_tbl)
+  task_id_cols <- subset_task_id_names(model_out_cols)
   grouped_model_outputs <- dplyr::group_by(model_out_tbl, dplyr::across(dplyr::all_of(c("model_id", task_id_cols))))
   terminal_output_type_id <- terminal_output_type_id[[terminal_output_type]]
 
-  if (terminal_output_type == "mean") {
-    converted_outputs_tmp <- dplyr::reframe(
-      grouped_model_outputs,
-      output_type_id = terminal_output_type_id,
-      value = mean(.data[["value"]])
-    )
-  } else if (terminal_output_type == "median") {
-    converted_outputs_tmp <- dplyr::reframe(
-      grouped_model_outputs,
-      output_type_id = terminal_output_type_id,
-      value = stats::median(.data[["value"]])
-    )
+  if (terminal_output_type %in% c("mean", "median")) {
+    otid_cols <- "output_type_id"
+    transform_fun <- get(terminal_output_type)
+    transform_args <- list(x = quote(.data[["value"]]))
+    grouped_model_outputs <- grouped_model_outputs
   } else if (terminal_output_type == "quantile") {
-    converted_outputs_tmp <- dplyr::reframe(
+    otid_cols <- "output_type_id"
+    transform_fun <- stats::quantile
+    transform_args <- list(
+      x = quote(unlist(.data[["initial_value"]])),
+      probs = quote(unlist(.data[["output_type_id"]])),
+      names = FALSE
+    )
```

Comment:
If you go for a non-nesting strategy, this would be what you would use this: 

```suggestion
    transform_args <- list(
      x = quote(.data[["value"]]),
      probs = quote(unique(.data[["output_type_id"]])),
      names = FALSE
    )
```